nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+
+```go
+type Registry interface {
+	WorkflowRegistry
+	ActivityRegistry
+}
+```
+
+Add and embed a third registry type: `OperationRegistry` with the following method:
+
+```go
+type OperationRegistry interface {
+	RegisterOperation(operation string, handler operation.Handler)
+	// Alternative: get the operation name from handler.Name()
+	// RegisterOperation(handler operation.Handler)
+}
+```
+
+`operation.Handler` is an interface with a minimal definition of:
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type Handler interface {
+	StartOperation(context.Context, *StartOperationRequest) (StartOperationResponse, error)
+	CancelOperation(context.Context, *CancelOperationRequest) error
+	// Later we will add GetResult and GetInfo as well
+}
+
+// Maps almost 1:1 to the Nexus Go SDK StartOperationRequest, exposing HTTP header and body directly instead of the HTTP
+// Request object.
+type StartOperationRequest struct {
+	Operation   string
+	RequestID   string
+	CallbackURL string
+
+	URL    *url.URL
+	Header http.Header
+	Body   []byte
+}
+
+// Maps almost 1:1 to the Nexus Go SDK CancelOperationRequest, exposing HTTP header and body directly instead of the
+// HTTP Request object.
+type CancelOperationRequest struct {
+	Operation   string
+	OperationID string
+
+	URL    *url.URL
+	Header http.Header
+}
+
+type OperationResponse interface {
+	// private methods
+}
+
+// Indicates that an operation completed successfully. Implements OperationResponse.
+type OperationResponseSync struct {
+	// Header to deliver in the HTTP response.
+	Header http.Header
+	// Body conveying the operation result.
+	// If it is an io.Closer, it will be automatically closed by the framework.
+	// Note that we accept a reader instead of bytes here to keep the future possiblity of supporting streaming of large
+	// payloads.
+	Body io.Reader
+}
+
+// Indicates that an operation has been accepted and will complete asynchronously. Implements OperationResponse.
+type OperationResponseAsync struct {
+	OperationID string
+}
+```
+
+To minimize the amount of boilerplate required to map operations to common Temporal constructs, the SDK exposes default
+handler implementations.
+
+### Workflow Run
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type WorkflowStartOptions struct {
+	// Options to start the workflow with. TaskQueue defaults to the current worker's task queue.
+	// The workflow ID used here will be returned as the operation ID by default.
+	client.WorkflowStartOptions
+	// Type of workflow to start.
+	WorkflowType string
+	// Input to provide to the workflow.
+	Input        []any
+	// TODO: Should Temporal Header be settable here?
+}
+
+// Returns a Handler that can maps and operation to a workflow that can be started and canceled.
+// Handles HTTP request to payload deserialization for input of type `T`.
+func NewWorkflowRunHandler[T any](func(context.Context, T, *StartOperationRequest) (*WorkflowStartOptions, error)) Handler
```

Comment:
Why `T`? I don't think it's too much to ask users to deserialize the request body themselves. You can add helpers on the start operation request if needed. But people need to be able to do this without args too.